dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -354,6 +352,19 @@ public Tool updateContainer(@ApiParam(hidden = true) @Parameter(hidden = true, n
             throw new CustomWebApplicationException("A published, private tool must have either an tool author email or tool maintainer email set up.", HttpStatus.SC_BAD_REQUEST);
         }
 
+        // An Amazon ECR repository cannot change its visibility once it's created. Thus, Amazon ECR tools cannot have their visibility changed.
+        if (registry == Registry.AMAZON_ECR) {
+            // Public Amazon ECR tool (public.ecr.aws docker path) can't be set to private
+            if (foundTool.getRegistry().equals(Registry.AMAZON_ECR.getDockerPath()) && tool.isPrivateAccess()) {
```

Comment:
Can `foundTool.getRegistry()` ever be null?  If yes, you should flip: `if (Registry.AMAZON_ECR.getDockerPath().equals(...`. If no, you may want to flip it anyway, just to be safe. :)